python/paddle/sparse/unary.py

```diff
@@ -920,3 +921,208 @@ def slice(x, axes, starts, ends, name=None):
+            N and M can be arbitraty positive number.
+            The data type of x should be float32 or float64.
+        q (int, optional): a slightly overestimated rank of :math:`X`.
+            Default value is :math:`q=min(6,N,M)`.
+        center (bool, optional): if True, center the input tensor.
+            Default value is True.
+        name (str, optional): Name for the operation (optional, default is None).
+            For more information, please refer to :ref:`api_guide_Name`.
+
+    Returns:
+        - Tensor U, is N x q matrix.
+        - Tensor S, is a vector with length q.
+        - Tensor V, is M x q matrix.
+
+        tuple (U, S, V): which is the nearly optimal approximation of a singular value decomposition of a centered matrix :math:`X`.
+
+    Examples:
+        .. code-block:: python
+
+            import paddle
+
+            format = "coo"
+            dense_x = paddle.randn((5, 5), dtype='float64')
+
+            if format == "coo":
+                sparse_x = dense_x.to_sparse_coo(len(dense_x.shape))
+            else:
+                sparse_x = dense_x.to_sparse_csr()
+
+            print("sparse.pca_lowrank API only support CUDA 11.x")
+            U, S, V = None, None, None
+            # use code blow when your device CUDA version >= 11.0
+            # U, S, V = paddle.sparse.pca_lowrank(sparse_x)
+
+            print(U)
+            # Tensor(shape=[5, 5], dtype=float64, place=Place(gpu:0), stop_gradient=True,
+            #        [[ 0.02206024,  0.53170082, -0.22392168, -0.48450657,  0.65720625],
+            #         [ 0.02206024,  0.53170082, -0.22392168, -0.32690402, -0.74819812],
+            #         [ 0.02206024,  0.53170082, -0.22392168,  0.81141059,  0.09099187],
+            #         [ 0.15045792,  0.37840027,  0.91333217, -0.00000000,  0.00000000],
+            #         [ 0.98787775, -0.09325209, -0.12410317, -0.00000000, -0.00000000]])
+
+            print(S)
+            # Tensor(shape=[5], dtype=float64, place=Place(gpu:0), stop_gradient=True,
+            #        [2.28621761, 0.93618564, 0.53234942, 0.00000000, 0.00000000])
+
+            print(V)
+            # Tensor(shape=[5, 5], dtype=float64, place=Place(gpu:0), stop_gradient=True,
+            #        [[ 0.26828910, -0.57116436, -0.26548201,  0.67342660, -0.27894114],
+            #         [-0.19592125, -0.31629129,  0.02001645, -0.50484498, -0.77865626],
+            #         [-0.82913017, -0.09391036,  0.37975388,  0.39938099, -0.00241046],
+            #         [-0.41163516,  0.27490410, -0.86666276,  0.03382656, -0.05230341],
+            #         [ 0.18092947,  0.69952818,  0.18385126,  0.36190987, -0.55959343]])
+    """
+
+    def get_floating_dtype(x):
+        dtype = x.dtype
+        if dtype in (paddle.float16, paddle.float32, paddle.float64):
+            return dtype
+        return paddle.float32
+
+    def matmul(x, B):
+        if x.is_sparse():
+            return paddle.sparse.matmul(x, B)
```

Comment:
This API requires sparse tensor as input(L1084) , so there is no need for this judgment and encapsulation anymore?

Reply:
some parts of code would use paddle.matmul such as L1057 `matmul(M, Q_c)`

Reply:
if `matmul(M, Q_c)` use `paddle.matmul` then write it directly, I think it is easier to understand.

Reply:
Done